src/ruleconfig.jl

```diff
@@ -0,0 +1,19 @@
+"""
+    ReverseRuleConfigBackend
+
+AD backend that uses reverse mode with any ChainRules-compatible reverse-mode AD package.
+"""
+struct ReverseRuleConfigBackend{RC <: RuleConfig} <: AbstractReverseMode
+    ruleconfig::RC
+end
+
+AD.@primitive function pullback_function(ab::ReverseRuleConfigBackend, f, xs...)
+    return (vs) -> begin
+        _, back = rrule_via_ad(ab.ruleconfig, f, xs...)
```

Comment:
What's reason for moving this in the function body? Isn't it better to obtain `back` only once outside of the function instead of doing it at every invocation of the pullback function?

Reply:
This is a good point.